commit 85cd553c3e56d9d12feb42bb1f6035710e5b40d1
Author: Carl Schwan <carl@carlschwan.eu>
Date:   Thu Jun 27 20:13:57 2024 +0200

    Setup statusbar and navigation color on Android
    
    Match the one of the header and the footer

diff --git a/style/qtquickcontrols/ApplicationWindow.qml b/style/qtquickcontrols/ApplicationWindow.qml
index 3f6aaf2..c7857ca 100644
--- a/style/qtquickcontrols/ApplicationWindow.qml
+++ b/style/qtquickcontrols/ApplicationWindow.qml
@@ -2,13 +2,46 @@
  * SPDX-License-Identifier: LGPL-2.1-only OR LGPL-3.0-only OR LicenseRef-KDE-Accepted-LGPL
  */
 
+import QtQml
 import QtQuick
 import QtQuick.Templates as T
 import org.kde.kirigami as Kirigami
+import org.kde.guiaddons as GuiAddons
 
 T.ApplicationWindow {
     id: window
-    // palette: Kirigami.Theme.palette
+
     Kirigami.Theme.colorSet: Kirigami.Theme.Window
     color: Kirigami.Theme.backgroundColor
+
+    Item {
+        id: headerColor
+
+        Kirigami.Theme.colorSet: Kirigami.Theme.Header
+        Kirigami.Theme.inherit: false
+
+        property color backgroundColor: Kirigami.Theme.backgroundColor
+        onBackgroundColorChanged: {
+            GuiAddons.WindowInsetsController.statusBarBackgroundColor = Kirigami.Theme.backgroundColor;
+        }
+    }
+
+    QtObject {
+        id: statusBarColor
+
+        readonly property color windowColor: window.Kirigami.Theme.backgroundColor
+        readonly property color footerColor: window.footer?.Kirigami.Theme.backgroundColor
+
+        onWindowColorChanged: {
+            if (!window.footer) {
+                GuiAddons.WindowInsetsController.navigationBarBackgroundColor = Kirigami.Theme.backgroundColor;
+            }
+        }
+
+        onFooterColorChanged: {
+            if (window.footer) {
+                GuiAddons.WindowInsetsController.navigationBarBackgroundColor = window.footer.Kirigami.Theme.backgroundColor;
+            }
+        }
+    }
 }
diff --git a/style/qtquickcontrols/CMakeLists.txt b/style/qtquickcontrols/CMakeLists.txt
index 2af77cf..0190808 100644
--- a/style/qtquickcontrols/CMakeLists.txt
+++ b/style/qtquickcontrols/CMakeLists.txt
@@ -6,6 +6,7 @@ ecm_add_qml_module(BreezeStyle URI "org.kde.breeze"
     "org.kde.kirigami"
     "org.kde.quickcharts"
     "org.kde.breeze.impl"
+    "org.kde.guiaddons"
 )
 
 ecm_target_qml_sources(BreezeStyle SOURCES
